dali/base/dadfs.cpp

```diff
@@ -1150,11 +1150,13 @@ protected: friend class CDistributedFile;
 // === Transactions
 class CDFAction: public CInterface
 {
-    unsigned locked;
+    unsigned locked = 0;
+    unsigned timeoutCount = 0;
 protected:
-    IDistributedFileTransactionExt *transaction;
+    IDistributedFileTransactionExt *transaction = nullptr;
     IArrayOf<IDistributedFile> lockedFiles;
-    DFTransactionState state;
+    DFTransactionState state = TAS_NONE;
+    StringBuffer tracing;
```

Comment:
Should really be a const StringAttr (though would need to add a printf-style constructor to StringAttr...)

Reply:
I'd prefer to leave that for another time/PR.
Would probably want to share some of the code that StringBuffer currently employs to check that the expanded printf string is not too big, and then to truncate after.